src/System.IO.Pipelines/MemoryPoolBlock.cs

```diff
@@ -39,13 +42,11 @@ protected MemoryPoolBlock(MemoryPool pool, MemoryPoolSlab slab, int offset, int
 
         public override int Length => _length;
 
-        public override Span<byte> Span
+        public override Span<byte> AsSpan(int index, int length)
         {
-            get
-            {
-                if (IsDisposed) PipelinesThrowHelper.ThrowObjectDisposedException(nameof(MemoryPoolBlock));
-                return new Span<byte>(Slab.Array, _offset, _length);
-            }
+            if (IsDisposed) PipelinesThrowHelper.ThrowObjectDisposedException(nameof(MemoryPoolBlock));
+            if (length > _length - index) throw new ArgumentOutOfRangeException();
```

Comment:
No, because _length is not the length of the whole buffer, but rather the length of the section represented by this OwnedBuffer.